custom_ops.py

```diff
@@ -0,0 +1,78 @@
+import torch
+import numpy as np
+# aka torch.library
+from library import OpDef, traceable, device_types
+
+# =====================================================================
+# This is example user library code. It defines
+# my_sin and my_sin_cos operators.
+# =====================================================================
+
+# User provides their custom op schema and implementations
+class MySin(OpDef):
+    schema = "(Tensor x) -> Tensor"
+
+    # the black-box cpu kernel
+    @staticmethod
+    def impl_cpu(x):
+        return torch.from_numpy(np.sin(x.detach().cpu().numpy()))
+
+    # the black-box cuda kernel
+    @staticmethod
+    def impl_cuda(x):
+        return torch.from_numpy(np.sin(x.detach().cpu().numpy())).to(x.device)
+
+    # the abstract impl. Must be "traceable". User must use opcheck to test.
+    @staticmethod
+    def abstract(x):
+        return torch.empty_like(x)
+
+    # autograd: provide us setup_backward() and backward() methods.
+    # these must be "traceable". User must use opcheck to test.
+    @staticmethod
+    def setup_backward(ctx, inputs, output):
```

Comment:
Would it be too cute to name this "forward" lol. Probably.....

Reply:
Is this only ever saving for backward? Would it be crazy to name this `save_for_backward`?